code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -490,33 +490,92 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	icon_state = "joint"
+	max_smoketime = 400
+	smoketime = 400
+	nicotine_amt = 0
+
+/obj/item/clothing/mask/smokable/cigarette/joint/blunt
+	name = "blunt"
+	desc = "A blunt lovingly rolled and crafted with care. Blaze it."
+	icon_state = "cigar"
 	max_smoketime = 500
 	smoketime = 500
-	nicotine_amt = 0
+	nicotine_amt = 4
 
-/obj/item/weapon/rollingpaper
+/obj/item/weapon/reagent_containers/rollingpaper
 	name = "rolling paper"
 	desc = "A small, thin piece of easily flammable paper, commonly used for rolling and smoking various dried plants."
 	description_fluff = "The legalization of certain substances propelled the sale of rolling papers through the roof. Now almost every Trans-stellar produces a variety, often of questionable quality."
 	icon = 'icons/obj/cigarettes.dmi'
 	icon_state = "cig paper"
 
-/obj/item/weapon/rollingpaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
+/obj/item/weapon/reagent_containers/rollingpaper/blunt
+	name = "blunt wrap"
+	desc = "A small piece of easily flammable paper similar to that which encases cigars. It's made out of tobacco, bigger than a standard rolling paper, and will last longer."
+	icon_state = "blunt paper"
+
+/obj/item/weapon/reagent_containers/rollingpaper/full
+	icon_state = "paper_full"
+
+/obj/item/weapon/reagent_containers/rollingpaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	if (istype(W, /obj/item/weapon/reagent_containers/food/snacks))
 		var/obj/item/weapon/reagent_containers/food/snacks/grown/G = W
 		if (!G.dry)
-			to_chat(user, "<span class='notice'>[G] must be dried before you roll it into [src].</span>")
+			to_chat(usr, "<span class='notice'>[G.name] must be dried before you add it to [src].</span>")
 			return
-		var/obj/item/clothing/mask/smokable/cigarette/joint/J = new /obj/item/clothing/mask/smokable/cigarette/joint(user.loc)
-		to_chat(usr, "<span class='notice'>You roll the [G.name] into a joint!</span>")
-		J.add_fingerprint(user)
+		var/obj/item/weapon/reagent_containers/rollingpaper/full/P = new /obj/item/weapon/reagent_containers/rollingpaper/full
+		P.reagents.maximum_volume = 25
+		if (istype(src, /obj/item/weapon/reagent_containers/rollingpaper/blunt))
+			P.name = "blunt wrap"
+			P.desc = "A small piece of easily flammable paper similar to that which encases cigars. It's made out of tobacco, bigger than a standard rolling paper, and will last longer."
+			P.icon_state = "blunt_full"
+			P.reagents.maximum_volume = 45
+		to_chat(usr, "<span class='notice'>You add the [G.name] to the [P.name].</span>")
+		P.add_fingerprint(user)
 		if(G.reagents)
-			G.reagents.trans_to_obj(J, G.reagents.total_volume)
-		J.name = "[G.name] joint"
-		J.desc = "A joint lovingly rolled and filled with [G.name]. Blaze it."
+			G.reagents.trans_to_obj(P, G.reagents.total_volume)
 		qdel(G)
+		user.put_in_hands(P)
+		user.drop_from_inventory(src)
+		qdel(src)
+
+/obj/item/weapon/reagent_containers/rollingpaper/full/attackby(obj/item/weapon/W as obj, mob/user as mob)

```

Comment:
How much of this proc is handled by the parent `/obj/item/weapon/reagent_containers/rollingpaper`'s `attackby()`? Because from what I can see this doesn't have any unique operations